core/templates/services/exploration-html-formatter.service.ts

```diff
@@ -68,37 +69,53 @@ export class ExplorationHtmlFormatterService {
       interactionCustomizationArgs: InteractionCustomizationArgs,
       parentHasLastAnswerProperty: boolean,
       labelForFocusTarget: string,
-      savedSolution: Solution): string {
+      savedSolution: string | undefined | null): string {
     var htmlInteractionId = this.camelCaseToHyphens.transform(interactionId);
     var element = $('<oppia-interactive-' + htmlInteractionId + '>');
 
     element = (
       this.extensionTagAssembler.formatCustomizationArgAttrs(
         element, interactionCustomizationArgs));
+    let tagEnd = '></oppia-interactive-' + htmlInteractionId + '>';
+    let directiveOuterHtml = element.get(0).outerHTML.replace(tagEnd, '');
+    let addSpace = true;
```

Comment:
Please avoid starting variable names with verbs.

Reply:
Curious, why not just add a space before and after each attribute by default instead of doing this bookkeeping with `addSpace`. Would that not work?

Reply:
> Curious, why not just add a space before and after each attribute by default instead of doing this bookkeeping with addSpace. Would that not work?

Becomes tricky with the unit tests.